consensus/raft/config.go

```diff
@@ -0,0 +1,212 @@
+// Config implements the ComponentConfig interface.
+type Config struct {
+	config.Saver
+
+	// A Hashicorp Raft's configuration object.
+	HashiraftCfg *hashiraft.Config
+	// A folder to store Raft's data.
+	DataFolder string
+}
+
+// ConfigJSON represents a human-friendly Config
+// object which can be saved to JSON.  Most configuration keys are converted
+// into simple types like strings, and key names aim to be self-explanatory
+// for the user.
+// Check https://godoc.org/github.com/hashicorp/raft#Config for extended
+// description on all Raft-specific keys.
+type jsonConfig struct {
+	// Storage folder for snapshots, log store etc. Used by
+	// the Raft.
+	DataFolder string `json:"data_folder,omitempty"`
+
+	// HeartbeatTimeout specifies the time in follower state without
+	// a leader before we attempt an election.
+	HeartbeatTimeout string `json:"heartbeat_timeout,omitempty"`
+
+	// ElectionTimeout specifies the time in candidate state without
+	// a leader before we attempt an election.
+	ElectionTimeout string `json:"election_timeout,omitempty"`
+
+	// CommitTimeout controls the time without an Apply() operation
+	// before we heartbeat to ensure a timely commit.
+	CommitTimeout string `json:"commit_timeout,omitempty"`
+
+	// MaxAppendEntries controls the maximum number of append entries
+	// to send at once.
+	MaxAppendEntries int `json:"max_append_entries,omitempty"`
+
+	// If we are a member of a cluster, and RemovePeer is invoked for the
+	// local node, then we forget all peers and transition into the
+	// follower state.
+	// If ShutdownOnRemove is is set, we additional shutdown Raft.
+	// Otherwise, we can become a leader of a cluster containing
+	// only this node.
+	ShutdownOnRemove bool `json:"shutdown_on_remove,omitempty"`
+
+	// TrailingLogs controls how many logs we leave after a snapshot.
+	TrailingLogs uint64 `json:"trailing_logs,omitempty"`
+
+	// SnapshotInterval controls how often we check if we should perform
+	// a snapshot.
+	SnapshotInterval string `json:"snapshot_interval,omitempty"`
+
+	// SnapshotThreshold controls how many outstanding logs there must be
+	// before we perform a snapshot.
+	SnapshotThreshold uint64 `json:"snapshot_threshold,omitempty"`
+
+	// LeaderLeaseTimeout is used to control how long the "lease" lasts
+	// for being the leader without being able to contact a quorum
+	// of nodes. If we reach this interval without contact, we will
+	// step down as leader.
+	LeaderLeaseTimeout string `json:"leader_lease_timeout,omitempty"`
+
+	// StartAsLeader forces Raft to start in the leader state. This should
+	// never be used except for testing purposes, as it can cause a split-brain.
+	StartAsLeader bool `json:"start_as_leader,omitempty"`
+
+	// The unique ID for this server across all time. When running with
+	// ProtocolVersion < 3, you must set this to be the same as the network
+	// address of your transport.
+	// LocalID string `json:local_id`
```

Comment:
Why is this being commented out?

Reply:
It's part of the new Raft version, it won't work right now.